__tests__/integration.test.js

```diff
@@ -140,3 +140,31 @@ describe('/api/articles/:article_id/comments', () => {
         })
     })
 })
+describe('/api/articles/:article_id/comments', () => {
+    test('POST: 201, creates a new comment for an article', () => {
+        const newComment = {
+            body: 'This is a test comment',
+            username: 'icellusedkars'
+        }
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.comment_id).toBe(19)
+            expect(response.body.comment.votes).toBe(0)
+            expect(response.body.comment.author).toBe('icellusedkars')
+            expect(response.body.comment.article_id).toBe(3)
+            expect(response.body.comment.body).toBe('This is a test comment')
+            expect(typeof response.body.comment.created_at).toBe('string')
+        })
+    })
+    test('POST: 400, sends an appropriate status and error message when given an invalid id', () => {
+        return request(app)
+        .post('/api/articles/not-an-article/comments')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('bad request')
+        })
+    })
```

Comment:
A solid happy path test 👍  you could use `.toMatchObject` or `.objectContaining` matcher if you wanted to save some typing. Also a great 400 for an invalid article id
It would be good to add another happy path test
- 201 ignores un-necessary properties on the comment body. e.g. 
```js
 const newComment = {
            body: 'This is a test comment',
            username: 'icellusedkars'
            extra: "ignores this"
        }
```
We're missing some sad path tests as well:
- 404 valid but non-existent article id eg. '/api/articles/9999/comments'
- 400 missing fields on the comment body (eg. an empty comment object)
- 404 username doesn't exist, ie. using a username that doesn't exist in the database